# Data Exploration - "The Office" Dataset

In [1]:
import pandas as pd

In [3]:
PATH = "../data/"
FILE = "the-office-lines - scripts.csv"

df = pd.read_csv(PATH+FILE, sep=",", index_col="id")
df.head()

season  episode  scene                                          line_text  \
id                                                                              
1        1        1      1  All right Jim. Your quarterlies look very good...   
2        1        1      1         Oh, I told you. I couldn't close it. So...   
3        1        1      1  So you've come to the master for guidance? Is ...   
4        1        1      1         Actually, you called me in here, but yeah.   
5        1        1      1    All right. Well, let me show you how it's done.   

    speaker  deleted  
id                    
1   Michael    False  
2       Jim    False  
3   Michael    False  
4       Jim    False  
5   Michael    False

In [4]:
df.shape

(59909, 6)

- The dataset includes 59909 rows (=lines spoken in the show) and 6 columns. 
- The first three columns indicate the season, episode and scene number of the line. 
- "line_text" contains the respective line in plain text. 
- The column "speaker" shows the name of the person who said the corresponding line. 
- Another column "deleted" indicates whether or not the corresponding line made it into the final version of the show.

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 59909 entries, 1 to 59909
Data columns (total 6 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   season     59909 non-null  int64 
 1   episode    59909 non-null  int64 
 2   scene      59909 non-null  int64 
 3   line_text  59909 non-null  object
 4   speaker    59909 non-null  object
 5   deleted    59909 non-null  bool  
dtypes: bool(1), int64(3), object(2)
memory usage: 2.8+ MB


- no missing values
- correct dtypes

In [7]:
df.describe(include="all")

season       episode         scene line_text  speaker deleted
count   59909.000000  59909.000000  59909.000000     59909    59909   59909
unique           NaN           NaN           NaN     52185      797       2
top              NaN           NaN           NaN     Yeah.  Michael   False
freq             NaN           NaN           NaN       486    12137   57973
mean        5.348178     11.558597     27.143852       NaN      NaN     NaN
std         2.389427      6.986208     17.860616       NaN      NaN     NaN
min         1.000000      1.000000      1.000000       NaN      NaN     NaN
25%         3.000000      5.000000     14.000000       NaN      NaN     NaN
50%         5.000000     11.000000     25.000000       NaN      NaN     NaN
75%         7.000000     18.000000     37.000000       NaN      NaN     NaN
max         9.000000     26.000000    116.000000       NaN      NaN     NaN

- 9 seasons with a maximum of 26 episodes with at most 116 scenes
- 52185 of the 59909 spoken lines are unique
- "Yeah." is the most frequently spoken line with a total of 486 occurrences
- there are 797 different people speaking throughout the show
- Michael is the character with the most lines with total number of 12137 lines
- 57973 of the available 59909 lines actually made it into the final version of the show